app/models/Test.php

```diff
@@ -568,4 +577,48 @@ public static function getSurveillanceData($from, $to)
 			return null;
 		}
 	}
+
+	/**
+	 * @param  Measure IDs $measureIds array()
+	 * @return Ranges whose interpretation is positive $positiveRanges array()
+	 */
+	public static function getPositiveRangesByMeasureIds($measureIds)
+	{
+		$positiveRanges = array();
+
+		foreach ($measureIds as $measureId) {
+
+			$measure = Measure::find($measureId);
+
+			$measureRanges = $measure->measureRanges;
+
+			foreach ($measureRanges as $measureRange) {
+
+				if ($measureRange->interpretation == '+') {
```

Comment:
Isn't '+' still hardcoding, ++ is still positive in some cases? So is +++ and ++++ e.t.c